bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/flamingos2/F2OiwfsProbeArm.scala

```diff
@@ -0,0 +1,111 @@
+package edu.gemini.spModel.gemini.flamingos2
+
+import java.awt.Shape
+import java.awt.geom.{Point2D, Rectangle2D}
+
+import edu.gemini.shared.util.immutable.ImPolygon
+import edu.gemini.skycalc.Offset
+import edu.gemini.spModel.core.{Angle, Coordinates}
+import edu.gemini.spModel.gemini.gems.Gems
+import edu.gemini.spModel.inst.{ArmAdjustment, ProbeArmGeometry}
+import edu.gemini.spModel.obs.context.ObsContext
+import edu.gemini.spModel.rich.shared.immutable._
+
+
+// The geometry is dependent on the plate scale determined by the choice of Lyot wheel,
+// and thus we must use a class instead of an object.
+object F2OiwfsProbeArm extends ProbeArmGeometry[Flamingos2] {
+  override protected val guideProbeInstance = Flamingos2OiwfsGuideProbe.instance
+
+  override def geometry(flamingos2: Flamingos2): List[Shape] = {
+    val probeArm: Shape = {
+      val plateScale = flamingos2.getLyotWheel.getPlateScale
+      val scaledLength = ProbePickoffArmLength * plateScale
+      val hm = PickoffMirrorSize * plateScale / 2.0
+      val htw = ProbeArmTaperedWidth / 2.0
+
+      val (x0, y0) = (hm, -htw)
+      val (x1, y1) = (x0 + ProbeArmTaperedLength, -hm)
+      val (x2, y2) = (x0 + scaledLength, y1)
+      val (x3, y3) = (x2, hm)
+      val (x4, y4) = (x1, y3)
+      val (x5, y5) = (x0, htw)
+
+      val points = List((x0, y0), (x1, y1), (x2, y2), (x3, y3), (x4, y4), (x5, y5))
+      ImPolygon(points)
+    }
+
+    val pickoffMirror: Shape = {
+      val plateScale = flamingos2.getLyotWheel.getPlateScale
+      val scaledMirrorSize = PickoffMirrorSize * plateScale
+      val xy = -scaledMirrorSize / 2.0
+      new Rectangle2D.Double(xy, xy, scaledMirrorSize, scaledMirrorSize)
+    }
+
+    List(probeArm, pickoffMirror)
+  }
+
+  override def armAdjustment(ctx0: ObsContext,
+                             guideStarCoords: Coordinates,
+                             offset0: Offset,
+                             T: Point2D): Option[ArmAdjustment] = {
+    import ProbeArmGeometry._
+
+    for {
+      ctx      <- Option(ctx0)
+      offset   <- Option(offset0)
+      gemsFlag = ctx.getAOComponent.asScalaOpt.fold(false){ ao =>
+        val aoNarrowType = ao.getNarrowType
+        aoNarrowType.equals(Gems.SP_TYPE.narrowType)
+      }
+    } yield {
```

Comment:
Since `gemsFlag` isn't used until the `yield`, maybe it should just appear there in the `yield` clause with all the other definitions?